webots_ros2_epuck/webots_ros2_epuck/simple_mapper.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 1996-2020 Cyberbotics Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from math import sin, cos
+import rclpy
+from rclpy.node import Node
+from rclpy.qos import DurabilityPolicy, HistoryPolicy, QoSProfile
+from nav_msgs.msg import OccupancyGrid
+from sensor_msgs.msg import LaserScan
+from geometry_msgs.msg import TransformStamped
+from tf2_ros import StaticTransformBroadcaster, TransformListener, Buffer
+from tf2_ros import LookupException, ConnectivityException, ExtrapolationException
+from builtin_interfaces.msg import Time
+from webots_ros2_core.math_utils import quaternion_to_euler
+
+
+WORLD_WIDTH = 3
+WORLD_HEIGHT = 3
+RESOLUTION = 0.01
+
+WORLD_ORIGIN_X = - WORLD_WIDTH / 2.0
+WORLD_ORIGIN_Y = - WORLD_HEIGHT / 2.0
+MAP_WIDTH = round(WORLD_WIDTH / RESOLUTION)
+MAP_HEIGHT = round(WORLD_HEIGHT / RESOLUTION)
+
+
+class SimpleMapper(Node):
+    def __init__(self, name):
+        super().__init__(name)
+
+        fill_map_param = self.declare_parameter('fill_map', True)
+
+        # Init map related elements
+        self.map = [-1]*MAP_WIDTH*MAP_HEIGHT
+        self.map_publisher = self.create_publisher(
+            OccupancyGrid,
+            '/map',
+            qos_profile=QoSProfile(
+                depth=1,
+                durability=DurabilityPolicy.TRANSIENT_LOCAL,
+                history=HistoryPolicy.KEEP_LAST,
+            )
```

Comment:
Just for the curiosity, why the default QoS profile was not working?

Reply:
I wanted to make it "latched" topic, so "late" subscribers get the last map from the buffer - they don't have to wait for the new map to be published.